Dockerfile

```diff
@@ -7,12 +7,12 @@ RUN apt-get update \
 COPY . /app 
 WORKDIR /app 
 ENV PATH="${PATH}:/root/.poetry/bin"
-RUN poetry install 
+RUN poetry config virtualenvs.create false --local && poetry install
 EXPOSE 5000
 
 FROM base as production
 ENV FLASK_ENV=production
-CMD ["poetry", "run", "gunicorn", "-b", "0.0.0.0:5000", "todo_app.app:create_app()"]
+CMD poetry run gunicorn -b 0.0.0.0:$PORT "todo_app.app:create_app()"
```

Comment:
This now actually *requires* a PORT env var to run, which adds a little complication to running it e.g. locally rather than on Heroku.

You can set a fallback value with `0.0.0.0:${PORT:-5000}` so it still works without $PORT set.